src/gsi/setupgnssrspd.f90

```diff
@@ -0,0 +1,951 @@
+                 miter = miter          )
+
+        if(.not.associated(my_diag)) call die(myname, &
+                'obsdiagLList_nextNode(), create =', .not.lobsdiag_allocated)
+     endif
+
+     if(.not.in_curbin) cycle
+
+!    Load obs error and u,v obs
+     obserror = max(cermin(ikx),min(cermax(ikx),data(ier,i)))
+     uob = data(iuob,i)
+     vob = data(ivob,i)
+ 
+ 
+     gnssrspdob=sqrt(uob*uob+vob*vob)
+     call tintrp2a1(ges_tv,tges,dlat,dlon,dtime,hrdifsig,&
+          nsig,mype,nfldsig)
+     call tintrp2a11(ges_ps,psges,dlat,dlon,dtime,hrdifsig,&
+          mype,nfldsig)
+     call tintrp2a1(ges_lnprsl,prsltmp,dlat,dlon,dtime,hrdifsig,&
+          nsig,mype,nfldsig)
+
+     factw = data(iff10,i)
+     if(sfcmod_gfs .or. sfcmod_mm5)then
+        sfcr = data(isfcr,i)
+        skint = data(iskint,i)
+        isli=data(idomsfc,i)
+        call comp_fact10(dlat,dlon,dtime,skint,sfcr,isli,mype,factw)
+     end if
+
+     nty=ictype(ikx)
+
+!    Process observations reported with height differently than those
+!    reported with pressure.  Type 260=nacelle 261=tower wind spd are
+!    encoded in NCEP prepbufr files with geometric height above
+!    sea level. 
+
+!    Process cygnss observations at mslp
+     if ( nty == 298) then
+          z_height = .true.
+          data(ihgt,i) = 10
+     endif
+     
+     if (z_height) then
+        
+        drpx = zero
+        dpres = data(ihgt,i)
+        dstn = data(istnelv,i)
+        call tintrp2a11(ges_z,zsges,dlat,dlon,dtime,hrdifsig,&
+             mype,nfldsig)
+
+!       Get guess surface elevation and geopotential height profile
+!       at observation location.
+        call tintrp2a1(geop_hgtl,zges,dlat,dlon,dtime,hrdifsig,&
+             nsig,mype,nfldsig)
+
+!       Convert observation height (in dpres) from meters to grid relative
+!       units.  Save the observation height in zob for later use.
+        zob = dpres
+        call grdcrd1(dpres,zges,nsig,1)
+        factw=one
+        rlow=zero
+        rhgh=zero
+
+!       Compute observation pressure (only used for diagnostics)
+!       Set indices of model levels below (k1) and above (k2) observation.
+        if (dpres<one) then
+           z1=zero;    p1=log(psges)
+           z2=zges(1); p2=prsltmp(1)
+        elseif (dpres>nsig) then
+           z1=zges(nsig-1); p1=prsltmp(nsig-1)
+           z2=zges(nsig);   p2=prsltmp(nsig)
+           drpx = 1.e6_r_kind
+        else
+           k=dpres
+           k1=min(max(1,k),nsig)
+           k2=max(1,min(k+1,nsig))
+           z1=zges(k1); p1=prsltmp(k1)
+           z2=zges(k2); p2=prsltmp(k2)
+        endif
+
+        dz21     = z2-z1
+        dlnp21   = p2-p1
+        dz       = zob-z1
+        pobl     = p1 + (dlnp21/dz21)*dz
+        presw    = ten*exp(pobl)
+
+!    Process observations with reported pressure
+     else
+
+        presw = ten*exp(dpres)
+        dpres = dpres-log(psges)
+        drpx=zero
+        if(nty >= 280 .and. nty < 290)then
```

Comment:
I thought from `read_gnssrspd.f90` that the obs type was 298.  Can it also fall in the range [280, 290)?

Reply:
I left this part of the code in case the finalized BUFR files end up containing GNSS-R wind observations with a reported pressure value.

Reply:
If the final BUFR files contain GNSS-R wind observations with  reported pressure value, wouldn't the type still be 298?

`setupgnssrspd.f90` has logic for `nty == 298` and `nty >= 280 .and. nty < 290`.  This doesn't completely align with `nty = 298` in `read_gnssrspd.f90`.   This is confusing.

Reply:
The numbers will have to be revisited or I could just delete this part for now.

Reply:
Shouldn't we finalize the `nty` values before we write code?  Doing so will likely also impact `read_gnssrspd.f90`.  It's odd to write code with specific type numbers when we don't know if these are the type numbers we'll actually use.

Reply:
I will delete these section. I don't know if NOAA will eventually have this information on their bufr files.

Reply:
@ilianagenkova FYI, in case obsproc ends up processing GNSS-R bufr files with pressure information, nty types will have to be assigned and setupgnssrspd.f90 and read_gnssrspd.f90 will need to be updated.

Reply:
@KariA-Spire, a few questions: 1. Is "nty" serving as "kx" for GNSS-R? 2How is the presence of reported surface pressure relevant to "nty"?
I

Reply:
@ilianagenkova 
Answers to your questions:
1. Yes
2. If there is a range of observations with specific reports of surface pressure, they would have to be assigned to nty or kx range
3. I changed it to 600 and also in read_gnssrspd.f90 per your ealier recommendation.: https://github.com/NOAA-EMC/GSI/pull/747#discussion_r1626645923